uwblueprint/child-witness-centre
# Install Chakra UI

## Notion Ticket Link

[Install Chakra UI](https://www.notion.so/uwblueprintexecs/Install-Chakra-UI-86ca298de7e940bc8a148c9001761e22)



## Implementation Description
* The actual installation changes are mostly in the frontend package.json and yarn.lock. The rest of the changes were just small fixes to accommodate the updated packages.



## Steps to Test
1. Delete your `frontend/node_modules` folder. In the frontend directory (`cd frontend`), run `yarn install` to get a fresh copy of the packages. This won't affect what Docker gives you, but it'll make the local package errors go away.
2. After running `docker compose down`, rebuild the containers with `docker compose up --build`.
3. Expected: app compiles with no errors*.

\* The login page will probably show "Script error" (and you'll see something about missing the client_id parameter in the console). Don't worry about it, the "Remove Google Auth for Login" ticket will make the error go away eventually.


## Screenshots (optional)
Here's what the Chakra UI button looks like:
![image](https://github.com/uwblueprint/child-witness-centre/assets/82423082/1668d905-3b46-43db-9376-db1539d60461)
If you want to also plop a component somewhere to try it out, check out the docs: https://chakra-ui.com/docs/components

## Checklist

## Linting & Formatting
### Mac
```bash
docker exec -it CWC-backend /bin/bash -c "yarn fix"
docker exec -it CWC-frontend /bin/bash -c "yarn fix"
```

### Windows
```bash
docker exec -it CWC-backend bash -c "yarn fix"
docker exec -it CWC-frontend bash -c "yarn fix"
```
